packages/taquito/src/contract/interface.ts

```diff
@@ -283,4 +284,34 @@ export interface ContractProvider extends StorageProvider {
    * @param params FailingNoop operation parameter
    */
   failingNoop(params: FailingNoopParams): Promise<FailingNoopOperation>;
+
+  /**
+   *
+   * @description Stake founds. Will sign and inject an operation using the current context
+   *
+   * @returns An operation handle with the result from the rpc node
+   *
+   * @param stake operation parameter
+   */
+  stake(params: StakingParams): Promise<TransactionOperation>;
+
+  /**
+   *
+   * @description Unstake founds. Will sign and inject an operation using the current context
```

Comment:
octez-client desc - `Unstake the given amount. If "everything" is given as amount, unstakes everything from the staking balance. Unstaked tez remains frozen for a set amount of cycles (the slashing period) after the operation. Once this period is over, the operation "finalize unstake" must be called for the funds to appear in the liquid balance.`